File: src/TitleCaseStrategy.ts
import { TextFormatterStrategy } from './TextFormatterStrategy';

export class TitleCaseStrategy implements TextFormatterStrategy {
  format(text: string): string {
    return text.replace(/\w\S*/g, (word) => {
      return word.charAt(0).toUpperCase() + word.substr(1).toLowerCase();
    });
  }
}

File: src/main/java/TitleCaseStrategy.java
public class TitleCaseStrategy implements TextFormatterStrategy {
    @Override
    public String format(String text) {
        String[] words = text.split("\\s+");
        StringBuilder titleCaseText = new StringBuilder();
        for (String word : words) {
            if (word.length() > 0) {
                titleCaseText.append(Character.toUpperCase(word.charAt(0)))
                        .append(word.substring(1).toLowerCase())
                        .append(" ");
            }
        }
        return titleCaseText.toString().trim();
    }
}
